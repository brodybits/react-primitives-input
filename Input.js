/* import TextInput from './imports' */
var TextInput = require('./TextInput')

/* import { createElement as h } from 'react' */
var h = require('react').createElement

var omit = require('object.omit');

var Input = (props) => {
  /* const { type, ...propsWithoutType } = props */
  var type = props.type
  var propsWithoutType = omit(props, 'type')

  /* return <TextInput {...propsWithoutType} /> */
  return h(TextInput, propsWithoutType)
}

/* export default Input */
module.exports = Input
