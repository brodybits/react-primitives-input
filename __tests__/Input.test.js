import React from 'react'

// XXX TBD XXX:
import Input from '../TextInput.js'

import renderer from 'react-test-renderer'

test('xxx', () => {
  // --
  const i1 = renderer.create(<Input/>)

  expect(i1.toJSON()).toMatchSnapshot()

})
