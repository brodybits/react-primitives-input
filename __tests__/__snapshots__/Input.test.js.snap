// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xxx 1`] = `
<input
  autoCapitalize="sentences"
  autoComplete="on"
  autoCorrect="on"
  className="css-textinput-1cwyjr8"
  data-focusable={true}
  dir="auto"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onSelect={[Function]}
  readOnly={false}
  spellCheck={true}
  type="text"
/>
`;
